default.nix

```diff
@@ -0,0 +1,155 @@
+# Use a specific version of nixpkgs from an bleeding-edge fork of github.com/NixOS/nixpkgs
+let
+  pkgs = import (fetchTarball
+    "https://github.com/rstats-on-nix/nixpkgs/archive/bc30fa6c937a4598564bd0fa7114cbad2a3554aa.tar.gz")
+    { };
+  # Add generic R packages required for this build
+  rpkgs = with pkgs.rPackages; [ BiocManager BiocBook ];
+
+  # Build mia package
+  mia = [
+    (pkgs.rPackages.buildRPackage {
+      name = "mia";
+      src = pkgs.fetchgit {
+        url = "https://github.com/microbiome/mia";
+        branchName = "devel";
+        rev = "58cc49bd118ff2b75d12b2eca006cfd48c307933";
+        sha256 = "sha256-BsQ7/nJBLadF2NQgGpHhAC10AFQaiK+HbMCIjfOH6d4=";
+      };
+      # mia dependencies (see DESCRIPTION)
+      propagatedBuildInputs = builtins.attrValues {
+        inherit (pkgs.rPackages)
+          SummarizedExperiment SingleCellExperiment TreeSummarizedExperiment
+          MultiAssayExperiment MASS ape decontam vegan BiocGenerics S4Vectors
+          IRanges Biostrings DECIPHER BiocParallel DelayedArray
+          DelayedMatrixStats scuttle scater DirichletMultinomial rlang dplyr
```

Comment:
Is this same file as cachix/default.nix?

Reply:
If it is possible, it is better to keep one in cachix directory since it makes root folder tidier

Reply:
Also, is it necessary to build the environment every time when PR gets new commits or is it enough that it is build when devel branch gets new commit?

The book is build only when devel receives new commit

Reply:
These files are identical, yes. Thanks for noting that.

It's not necessary to build the environment every time actually. It's pinned to a particular commit at the moment, and that one needs to be manually updated every now and then if we want to updated packages and R version. It's also required to rebuild it if we add new packages and/or system dependencies.

I can completely remove that and set it to be manually run. Should be possible with

```yaml
on:
  workflow_dispatch:
```

Reply:
The location of `default.nix` can be in `cachix`. Then, to enter the shell, users have to go to that directory, run `nix-shell`, and then exit it to have access to the entire `OMA` repo.